dev/release/check-license.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This script checks if all the files mentioned in the LICENSE file are present in the project.
+# Besides highlighting the missing files, it will display possible locations for any new files.
+# It adds text like [NOT FOUND] --> "new/path/to/the/file" in the end of the line where the file is missing.
+# With the output of this script, you can update the LICENSE file with the correct file paths.
+
+FAILED=0
+
+PROJECT_ROOT=$(dirname $(dirname $(dirname $(readlink -f "$0"))))
```

Comment:
Can you please follow the `update-java-doc-version.sh` to let user can also specify the project dir, like:

```
PROJECT_DIR=${2:-$(cd "$(dirname "$0")" && pwd)/../../}
```

Reply:
Besides, did you verify both on Linux and macOS. My concern is that some commands may not be compatible between Linux and Mac.

Reply:
Hi,  I've adjusted as requested and ensure the update works on both Mac and Linux. 
I think the "#!/usr/bin/env bash" can provide the best compatibility across different systems

Below is the test result on ubuntu 22.04
<img width="1187" alt="Screenshot 2024-11-13 at 1 43 44 PM" src="https://github.com/user-attachments/assets/57bb0112-6934-4890-9064-69f0b355dca5">